### PR TITLE
Replace Flask-Sockets with aiohttp for testing 

### PR DESCRIPTION
**Pull request by WilliamBergamin**
This PR bring the logic from [slack-sdk #1445](https://github.com/slackapi/python-slack-sdk/pull/1445) to bolt python

`Flask-Sockets` [is deprecated](https://github.com/heroku-python/flask-sockets/issues/85) and it won't receive any fixes in the future, so it seems reasonable to replace it with some up-to-date library.
`aiohttp` has been chosen since it was already presented as an optional requirement.

### Category (place an `x` in each of the `[ ]`)

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [ ] Adapters in `slack_bolt.adapter`
* [ ] Document pages under `/docs`
* [x] Others

## Requirements (place an `x` in each `[ ]`)

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
         run: |
           pip install -U pip
           pip install -r requirements.txt
-          pip install -r requirements/testing_without_asyncio.txt
```
Comment (seratch) [inline .github/workflows/tests.yml:26]: Can you revert this change? This part intentionally does not have aiohttp to detect unexectedly having aiohttp depedency to the core part  (as the following name "Run tests without aiohttp" indicate)

**Review by seratch — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 1 above)
Comment (WilliamBergamin) [reply .github/workflows/tests.yml]: I've reverted this change but `aoihttp` will be required to run `pytest tests/adapter_tests/socket_mode/`, we can try using another synchronous library or move the CI step `Run tests for Socket Mode adapters` after `Run tests for HTTP Mode adapters (asyncio-based libraries)` let me know what you think

Comment (seratch) [reply .github/workflows/tests.yml]: > pytest tests/adapter_tests/socket_mode/

Sorry, I should have clearly mentioned this but adding aiohttp for these tests is totally. My concern was about tests under slack_sdk etc.

Comment (WilliamBergamin) [reply .github/workflows/tests.yml]: Ohh I see, I've update the ci tests and dependencies to make them pass, let me know if we prefer a sync approach to this?

Comment (seratch) [reply .github/workflows/tests.yml]: Thank you! The only risk is that we may unintentionally add aiohttp imports to the non-asyncio Socket Mode adapters. Thus, making the changes only to the adapter tests should be safe enough. We can detect such in code reviews.

**Review by seratch — APPROVED** [review]
